### ISSUE TITLE
iOS 17 Support

### ISSUE THREAD

**Issue by chrisvasselli** [shinydevelopment/SimulatorStatusMagic#115]
It looks like iOS 17 has removed the `UIStatusBarServerListener` class which contains the override struct. It seems like the current approach used by SimulatorStatusMagic may have been eliminated in iOS 17. I'll take another stab at finding an alternative, but for now I'd say it's likely that iOS 16 will be the last version supported by SimulatorStatusMagic.

If anyone is interested in helping me investigate, you can dump the relevant iOS 17 headers by attaching the iOS 17 sim runtime at `/Library/Developer/CoreSimulator/Images` (for beta 2, it's 2A5F1E26-3761-41C2-87C1-6F8415AF2C63.dmg), and then use [ipsw](github.com/blacktop/ipsw) to dump the headers for UIKitCore:

`ipsw macho info -a arm64 -o /Library/Developer/CoreSimulator/Volumes/iOS_21A5248u/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS\ 17.0.simruntime/Contents/Resources/RuntimeRoot/System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore > ~/Desktop/uikitcore.txt`

**Comment by chrisvasselli** [comment]
I took another crack at this. It looks like use of the `UIStatusBarServer` API is now restricted to the SpringBoard process. When you try to use it, you get an error message printed: "Unsupported use of UIStatusBarSever SPI outside of SpringBoard. Use System Status APIs instead.".

There is a private framework called SystemStatus.framework, so I am guessing that's what it's referring to. I spent some time trying to see if I could reverse engineer how that works, but it seems very different from the UIStatusBarServer API, and involves XPC connections that it is not obvious to me how to configure.

I haven't given up hope completely, but it is still feeling like SimulatorStatusMagic will probably not work on iOS 17 without a lot of work.
